pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -1174,6 +1180,26 @@ private static void handleHLLLog2mOverride(PinotQuery pinotQuery, int hllLog2mOv
     }
   }
 
+  private boolean isDisableGroovy(@Nullable TableConfig offlineTableConfig, @Nullable TableConfig realtimeTableConfig) {
+    Boolean offlineTableDisableGroovyQuery = null;
+    if (offlineTableConfig != null && offlineTableConfig.getQueryConfig() != null) {
+      offlineTableDisableGroovyQuery = offlineTableConfig.getQueryConfig().getDisableGroovyQuery();
+    }
+
+    Boolean realtimeTableDisableGroovyQuery = null;
+    if (realtimeTableConfig != null && realtimeTableConfig.getQueryConfig() != null) {
+      realtimeTableDisableGroovyQuery = realtimeTableConfig.getQueryConfig().getDisableGroovyQuery();
+    }
+
+    if (offlineTableDisableGroovyQuery == null && realtimeTableDisableGroovyQuery == null) {
+      return _disableGroovy;
+    }
+
+    // If offline or online table config disables Groovy, then Groovy should be disabled
+    return (offlineTableDisableGroovyQuery == null ? false : offlineTableDisableGroovyQuery) || (
+        realtimeTableDisableGroovyQuery == null ? false : realtimeTableDisableGroovyQuery);
```

Comment:
(minor) Can be simplified to 
```suggestion
    return Boolean.TRUE.equals(offlineTableDisableGroovyQuery) || Boolean.TRUE.equals(realtimeTableDisableGroovyQuery);
```